src/components/ResourceMachine.tsx

```diff
@@ -1,17 +1,36 @@
 import React from 'react';
 import {Text, View, TouchableOpacity} from 'react-native';
+import {Machine, ResourceType} from '../types';
+import GlobalStyles from '../styles/GlobalStyles';
+import ResourceStyles from '../styles/ResourceStyles';
 
-const ResourceMachine = () => {
+const getCosts = (costs: {[x in ResourceType]?: number}) => {
   return (
-    <View>
-      <Text>Miner: 2</Text>
-      <Text>Build a pickaxe for your miner.</Text>
-      <Text>Cost:</Text>
-      <Text>Input:</Text>
-      <Text>Output:</Text>
-      <TouchableOpacity>
-        <Text>Get 1</Text>
-      </TouchableOpacity>
+    <View style={ResourceStyles.machineBox}>
+      <Text style={GlobalStyles.property}>Cost:</Text>
+    </View>
+  );
+};
+
+const ResourceMachine = ({machine}: {machine: Machine}) => {
+  return (
+    <View style={ResourceStyles.machineCard}>
+      <View>
+        <Text style={GlobalStyles.header}>{machine.name}</Text>
+        <Text>{machine.desc}</Text>
+      </View>
+      <View style={ResourceStyles.machineContainer}>
+        {getCosts(machine.cost)}
```

Comment:
I haven't really thought about this yet but I was thinking of doing something like:

```
{
    Object.keys(machine.cost).map(resourceCost => <ResourceCost {SOME_PROPS}/>)
}
```

The main advantage being that there's some logic on how to actually display this which gets reused like this for anything with a cost :) I'm even wondering if it might be best to create a new `ConnectedResourceCost` component which takes a `ResourceType` (e.g. 'metal') and `currentCost` as a parameter and gets connected with redux if that makes sense. Sort of a HOC for this to simplify logic everywhere